flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -449,12 +364,15 @@ public void dispose() {
 
 			cleanInstanceBasePath();
 		}
+		this.disposed = true;
 	}
 
 	private static void addColumnFamilyToCloseLater(
 		List<ColumnFamilyOptions> columnFamilyOptions, ColumnFamilyHandle columnFamilyHandle) {
-
 		try {
+			if (columnFamilyHandle == null || columnFamilyHandle.getDescriptor() == null) {
```

Comment:
why do we suddenly need those checks? I would assume that all arguments are never null, the code should be null-free. If there is not a good reason why we need the check, I would remove it because it can make you doubt if the arguments _might_ be null.

Reply:
Mainly for testing, like if we use a mocked `ColumnFamilyHandle` as the `injectedDefaultColumnFamilyHandle` for builder.

Reply:
I think it would be good if we can avoid introducing null-checks in production code that are only needed for tests. I would suggest to rather ensure that the test does not pass in nulls.